src/js/ripple/keypairs.js

```diff
@@ -0,0 +1,283 @@
+  this.publicKey = keyPair.publicKey;
+  this.type = KeyType.ed25519;
+}
+
+util.inherits(Ed25519Pair, KeyPair);
+
+hasCachedProperty(Ed25519Pair.prototype, 'pubKeyBytes', function() {
+  return [0xED].concat(toGenericArray(this.publicKey));
+});
+
+Ed25519Pair.prototype.sign = function(message) {
+  var messageArray = new Uint8Array(message);
+  var signatureArray = nacl.sign.detached(messageArray, this.secretKey);
+  return toGenericArray(signatureArray);
+};
+
+Ed25519Pair.prototype.verify = function(message, signature) {
+  var messageArray = new Uint8Array(message);
+  var signatureArray = new Uint8Array(signature);
+  return nacl.sign.detached.verify(messageArray, signatureArray,
+                                   this.publicKey);
+};
+
+/* ---------------------------- SECP256K1 KEYPAIR --------------------------- */
+
+/*
+* @class
+* @private
+*/
+function Secp256k1Pair() {
+  KeyPair.apply(this, arguments);
+  this.keyPair = this.seed.get_key();
+  this.type = KeyType.secp256k1;
+}
+
+util.inherits(Secp256k1Pair, KeyPair);
+
+hasCachedProperty(Secp256k1Pair.prototype, 'pubKeyBytes', function() {
+  return hexToArray(this.keyPair.to_hex_pub());
+});
+
+/*
+@param {Array<Number>} message (bytes)
+ */
+Secp256k1Pair.prototype.sign = function(message) {
+  var hash = this.hashMessage(message);
+  var signatureBits = this.keyPair.sign(hash, 0);
+  return sjcl.codec.bytes.fromBits(signatureBits);
+};
+
+/*
+@param {Array<Number>} message (bytes)
+@return {UInt256} 256 bit hash of the message
+ */
+Secp256k1Pair.prototype.hashMessage = function(message) {
+  var bits = sjcl.codec.bytes.toBits(message);
+  return Crypt.hashSha512Half(bits);
+};
+
+/*
+@param {Array<Number>} message (bytes)
+@param {Array<Number>} signature (bytes)
+ */
+Secp256k1Pair.prototype.verify = function(message, signature) {
+  var sig = signature;
+  var rPos = 4;
+  var rLen = sig[3];
+  var sPos = rLen + 6;
+  var sLen = sig[rLen + 5];
+  var r = sig.slice(rPos, rPos + rLen);
+  var s = sig.slice(sPos, sPos + sLen);
+  var rs = sjcl.bitArray.concat(bytesBnTo256Bits(r),
+                                bytesBnTo256Bits(s));
+  try {
+    return this.keyPair._pub().verify(this.hashMessage(message).to_bits(), rs);
+  } catch (e) {
+    return false;
+  }
+};
+
+exports.getKeyPair = function(specifier) {
+  if (specifier instanceof KeyPair) {
+    return specifier;
+  }
+
+  var isGeneric = typeof specifier === 'string';
+  var specifierObj = isGeneric ? {} : specifier;
+  var keyType = specifierObj.key_type;
+
+  if (isGeneric) {
+    specifierObj.generic = arguments[0];
+  }
+  if (!keyType) {
+    keyType = KeyType.secp256k1;
+  }
+
+  var seed = parseSeed(specifierObj);
+
+  if (keyType === KeyType.secp256k1) {
+    return new Secp256k1Pair(seed, specifier);
+  } else if (keyType === KeyType.ed25519) {
+    return new Ed25519Pair(seed, specifier);
+  }
+
+  throw new Error('unknown key_type ' + keyType);
+};
+
+exports.keyPairCache = {};
```

Comment:
I think we could probably remove the caching from this pull request, as it needs more consideration, for various reason.